modules/siddhi-extensions/input-mappers/json-input-mapper/src/main/java/org/wso2/siddhi/extension/input/mapper/json/JsonSourceMapper.java

```diff
@@ -0,0 +1,515 @@
+                                        parser.getText() + " is not compatible with type FLOAT");
+                            }
+                        case LONG:
+                            if (JsonToken.VALUE_NUMBER_INT.equals(jsonToken)) {
+                                data[position] = parser.getValueAsLong();
+                                break;
+                            } else {
+                                throw new ExecutionPlanRuntimeException("Json message " + eventObject.toString() +
+                                        " contains incompatible attribute types and values. Value " +
+                                        parser.getText() + " is not compatible with type LONG");
+                            }
+                        default:
+                            return null;
+                    }
+                }
+            }
+        }
+
+        if (failOnMissingAttribute && (numberOfProvidedAttributes != attributesSize)) {
+            throw new ExecutionPlanRuntimeException("Json message " + eventObject.toString() +
+                    " contains missing attributes. Hence dropping the message.");
+        }
+        return event;
+    }
+
+    private Event[] convertToEventArrayForDefaultMapping(Object eventObject) {
+        Gson gson = new Gson();
+        JsonObject[] eventObjects = gson.fromJson(eventObject.toString(), JsonObject[].class);
+        Event[] events = new Event[eventObjects.length];
+        int index = 0;
+        JsonObject eventObj = null;
+        for (JsonObject jsonEvent : eventObjects) {
+            if (jsonEvent.has(DEFAULT_JSON_EVENT_IDENTIFIER)) {
+                eventObj = jsonEvent.get(DEFAULT_JSON_EVENT_IDENTIFIER).getAsJsonObject();
+                if (failOnMissingAttribute && eventObj.size() < streamAttributes.size()) {
+                    log.error("Json message " + eventObj.toString() + " contains missing attributes. " +
+                            "Hence dropping the message.");
+                    continue;
+                }
+            } else {
+                log.error("Default json message " + eventObj.toString()
+                        + " in the array does not have the valid event identifier \"event\". " +
+                        "Hence dropping the message.");
+                continue;
+            }
+            Event event = new Event(streamAttributes.size());
+            Object[] data = event.getData();
+
+
+            int position = 0;
+            for (Attribute attribute : streamAttributes) {
+                String attributeName = attribute.getName();
+                Attribute.Type type = attribute.getType();
+                Object attributeValue = eventObj.get(attributeName);
+                if (attributeValue == null) {
+                    data[position++] = null;
+                } else {
+                    data[position++] = attributeConverter.getPropertyValue(
```

Comment:
Single event processing has been implemented using jackson tokenizer and default multiple events processing has been implemented using gson.